src/Client/MainComponents/SpreadsheetView/DataMap.fs

```diff
@@ -0,0 +1,123 @@
+    let cell = model.SpreadsheetModel.DataMapOrDefault.GetCell(columnIndex, rowIndex)
+    [
+        Spreadsheet.Cells.Cell.Body (index, cell, model, dispatch)
+        let isExtended = state.Contains columnIndex
+        let header = Spreadsheet.Controller.Generic.getHeader columnIndex model.SpreadsheetModel
+        if (cell.isTerm || cell.isUnitized) && isExtended then
+            if cell.isUnitized then 
+                Cell.BodyUnit(index, cell, model, dispatch)
+            else
+                Cell.Empty()
+            Cell.BodyTSR(index, cell, model, dispatch)
+            Cell.BodyTAN(index, cell, model, dispatch)
+        elif header.IsDataColumn then
+            if cell.isData then
+                Cell.BodyDataSelector(index, cell, model, dispatch)
+                Cell.BodyDataFormat(index, cell, model, dispatch)
+                Cell.BodyDataSelectorFormat(index, cell, model, dispatch)
+            else
+                Cell.Empty()
+                Cell.Empty()
+                Cell.Empty()
+
+    ]
+
+let private CreateHeaderCells(columnIndex, state, setState, model, dispatch) =
+    let header = DataMap.getHeader columnIndex
+    [
+        Spreadsheet.Cells.Cell.Header(columnIndex, header, state, setState, model, dispatch, readonly = true)
+        if header.IsTermColumn then
+            let isExtended = state.Contains columnIndex
+            if isExtended then
+                Cell.HeaderUnit(columnIndex, header, state, setState, model, dispatch)
+                Cell.HeaderTSR(columnIndex, header, state, setState, model, dispatch)
+                Cell.HeaderTAN(columnIndex, header, state, setState, model, dispatch)
+        elif header.IsDataColumn then
+            Cell.HeaderDataSelector(columnIndex, header, state, setState, model, dispatch)
+            Cell.HeaderDataFormat(columnIndex, header, state, setState, model, dispatch)
+            Cell.HeaderDataSelectorFormat(columnIndex, header, state, setState, model, dispatch)
+        else
+            ()
+    ]
+
+let private BodyRow (rowIndex: int) (state:Set<int>) (model:Model) (dispatch: Msg -> unit) =
+    [
+        //MainComponents.CellStyles.RowLabel rowIndex
+        //Spreadsheet.Cells.Cell.BodyBase(
+        //    Spreadsheet.ColumnType.Main,
+        //    model.SpreadsheetModel.DataMapOrDefault.DataContexts.[rowIndex].Name |> Option.defaultValue "",
+        //    (fun _ -> ()),
+        //    (NameIndex, rowIndex),
+        //    model,
+        //    dispatch,
+        //    readonly=true,
+        //    tooltip="This field is calculated from `Data File Path` and `Data Selector`"
+        //)
+        for columnIndex in 0 .. (model.SpreadsheetModel.DataMapOrDefault.ColumnCount-1) do
+            (
+                columnIndex,
+                rowIndex,
+                state,
+                model,
+                dispatch
+            )    
+    ]
+
+let private BodyRows (state:Set<int>) (model:Model) (dispatch: Msg -> unit) =
```

Comment:
Same as above, naming of the function is not helpful

Reply:
Explained in comment for `BodyRow`

Reply:
see above